### ISSUE TITLE
Just Refactor it, damnit!!

### ISSUE THREAD

**Issue by DronRathore** [DronRathore/goexpress#25]
Need to refactor this repo, sorry for being a slacker and not maintaining it for quite while, refactor will add support for context, and coding styles to make it more *Go* like(obviously a breaking change).

Looking forward to the Christmas holidays to finish the refactor. If you want some *must to have* features, feel free to add them here.

**Comment by ilyakaznacheev** [comment]
Just `gofmt` it.
Actually, tabs are a [Go standard](https://blog.golang.org/go-fmt-your-code)

**Comment by DronRathore** [comment]
Yes I am just more accustomed with spaces in my day-to-day work hence preferred that in comparison to the time when I initially wrote the library.
